### PR TITLE
LCH-6356: Adding guzzle 8 compatibility. 2.x.

### PR DESCRIPTION
**Pull request by narendradesai**
Issue: https://backlog.acquia.com/browse/LCH-6356

### CODE REVIEW

File: src/ContentHubClient.php [hunk 1]
```diff
@@ -26,7 +29,10 @@
  *
  * @package Acquia\ContentHubClient
  */
-class ContentHubClient extends Client {
+class ContentHubClient implements ClientInterface {
+
+  use ClientTrait;
```
Comment (saxumVermes) [inline src/ContentHubClient.php]: ClientTrait is not available in guzzle <7.0, however, Drupal 9.x supports guzzle 6. In this particular case the best would be to actually just implement the methods, get, put, delete, post and use the client we already have to execute those requests.

File: src/ContentHubClientTrait.php [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+use GuzzleHttp\Promise\PromiseInterface;
+use GuzzleHttp\Psr7\Response;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LogLevel;
+
+/**
+ * Common trait for CH Client and CH Logging Client.
+ */
+trait ContentHubClientTrait {
```
Comment (saxumVermes) [inline src/ContentHubClientTrait.php:16]: The implementation of get, put, post, delete can go here and we can use httpClient to carry out those requests.

**Comment by narendradesai** [comment]
CH 2.x module PR: https://github.com/acquia/acquia_contenthub/pull/962

**Review by saxumVermes — APPROVED** [review]
Looks great!

**Review by amangrover90 — APPROVED** [review]
LGTM!!

**Review by saxumVermes — APPROVED** [review]
Looks good!

**Pull request merged**